File: backend/Persistence/Models/PropertyLocation.cs
using System;
using System.Collections.Generic;

namespace Persistence.Models;

public partial class PropertyLocation
{
    public Guid Id { get; set; }

    public string StreetName { get; set; } = null!;

    public string HouseNumber { get; set; } = null!;

    public string City { get; set; } = null!;

    public string Country { get; set; } = null!;

    public string? ZipCode { get; set; }

    public double Longitude { get; set; }

    public double Latitude { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual ICollection<Property> Properties { get; set; } = new List<Property>();
}

File: mobile/lib/src/features/dashboard/domain/entities/property_location.dart
import 'package:equatable/equatable.dart';

class PropertyLocation extends Equatable {
  final String id;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String streetName;
  final String houseNumber;
  final String city;
  final String country;
  final String zipCode;
  final double longitude;
  final double latitude;

  const PropertyLocation({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.streetName,
    required this.houseNumber,
    required this.city,
    required this.country,
    required this.zipCode,
    required this.longitude,
    required this.latitude,
  });

  @override
  List<Object?> get props => [
        id,
        createdAt,
        updatedAt,
        streetName,
        houseNumber,
        city,
        country,
        zipCode,
        longitude,
        latitude,
      ];
}
